astropy/convolution/tests/test_convolve_fft.py

```diff
@@ -178,15 +178,10 @@ def test_unity_3_withnan(self, boundary, nan_treatment, normalize_kernel,
             assert np.isnan(z[1])
 
         z = np.nan_to_num(z)
-
-        # for whatever reason, numpy's fft has very limited precision, and
-        # the comparison fails unless you cast the float64 to a float16
-        if hasattr(np, 'float16'):
-            assert_array_almost_equal_nulp(
-                np.asarray(z, dtype=np.float16),
-                np.array([1., 0., 3.], dtype=np.float16), 10)
-        # ASSERT equality to better than 16 bit but worse than 32 bit precision
-        assert_allclose(z, np.array([1., 0., 3.]), atol=1e-14)
+        # ASSERT equality to better to almost 64 bit precision
+        # (relative to the of-order-unity power in the FFTs; i.e.,
+        # i.e., 10 ULP = 10*np.finfo(np.float64).eps = 2e-15).
```

Comment:
Right, this makes sense.

Would it be better to just do:
`assert_allclose(z, np.array([1.,0.,3.]), atol=10*np.finfo(np.float64.eps)`
or even
`np.testing.assert_array_almost_equal_nulp(z, arr, 10)` ?
(it's likely that if/when I wrote the original test, I wasn't aware of nulp testing...)

Reply:
FOLLOWUP: ignore that, I now understand that the original problem is that the nulp testing in numpy doesn't work, and you're removing my nulp testing.  I just didn't read carefully enough.

Reply:
Well, the `nulp` testing works, but it is not correct for our case: the numbers are not within a few ULP from zero because it involved fourier amplitudes near 1.